crypto/libsodium-fork/src/libsodium/crypto_core/ed25519/ref10/ed25519_ref10.c

```diff
@@ -982,6 +1152,32 @@ ge25519_is_on_main_subgroup(const ge25519_p3 *p)
     return fe25519_iszero(pl.X);
 }
 
+int
+ge25519_is_canonical_vartime(const unsigned char *s)
```

Comment:
I'm pretty confident that this implementation could be improved by avoiding using the memcmp, and making explicit comparison. On modern 64-bit cpu, we really need only 4 comparison for each of the addresses.